beamer/agent.py

```diff
@@ -31,107 +37,154 @@ def _get_deployment_block(contract_info: dict[str, ContractInfo]) -> BlockNumber
     return min(request_manager_deployment_block, fill_manager_deployment_block)
 
 
+@dataclass
+class Chain:
+    w3: Web3
+    id: ChainId
+    rpc_url: URL
+
+
+@dataclass
+class Rollup(Chain):
+    contracts_info: dict[str, ContractInfo]
+    contracts: dict[str, Contract]
+    request_manager: RequestManager
+    fill_manager: FillManager
```

Comment:
Both of these two dataclass types should be private, because they won't be used outside of this module.

Reply:
nitpick: Do we really need to store the request_manager and fill_manager on the class level? A property annotation could be enough as they are stored in contracts, aren't they?